src/lib/components/Search/index.js

```diff
@@ -7,18 +7,28 @@ import {BaseElement} from '../BaseElement';
 import {store} from '../../store';
 import * as router from '../../utils/router';
 import {debounce} from '../../utils/debounce';
-import algoliasearch from 'algoliasearch/dist/algoliasearch-lite.esm.browser';
 import {trackError} from '../../analytics';
 import 'focus-visible';
 import './_styles.scss';
 
-// Create an algolia client so we can get search results.
-// These keys are safe to be public.
-const applicationID = '2JPAZHQ6K7';
-const apiKey = '01ca870a3f1cad9984ed72419a12577c';
-const indexName = 'webdev';
-const client = algoliasearch(applicationID, apiKey);
-const index = client.initIndex(indexName);
+let algoliaSearchIndex;
+
+async function loadAlgoliaLibrary() {
```

Comment:
Doing two searches in rapid succession would result in two `import()` statements being called, and two `algoliasearch` things being created.

Reply:
This still isn't fixed.

My suggestion is to have two methods:

1. `internalLoadAlgoliaLibrary` which does everything, doesn't affect global state, and returns the result of `client.initIndex(...)`

2. `loadAlgoliaLibrary`, which _either_ sets `algoliaIndexPromise` to the result of the above method, or returns that result.

Basically if you call `loadAlgoliaLibrary` lots and lots of times you won't cause further requests.

Reply:
Addressed in https://github.com/GoogleChrome/web.dev/pull/4175/commits/e7b8efa76b69b4c145ec153c97b11cd04333d678